runtime/src/main/java/org/corfudb/runtime/collections/streaming/DeltaStream.java

```diff
@@ -0,0 +1,184 @@
+package org.corfudb.runtime.collections.streaming;
+
+import com.google.common.base.Preconditions;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.Buffer;
+import org.apache.commons.collections.BufferUtils;
+import org.apache.commons.collections.buffer.BoundedFifoBuffer;
+import org.corfudb.common.metrics.micrometer.MicroMeterUtils;
+import org.corfudb.protocols.wireprotocol.ILogData;
+import org.corfudb.runtime.exceptions.TrimmedException;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.runtime.view.AddressSpaceView;
+import org.corfudb.runtime.view.ReadOptions;
+import org.corfudb.runtime.view.stream.StreamAddressSpace;
+
+import java.time.Duration;
+import java.util.UUID;
+
+/**
+ * Unlike the {@link org.corfudb.runtime.view.stream.IStreamView} this stream implementation only tracks updates
+ * that haven't been checkpointed.
+ */
+@Slf4j
+public class DeltaStream {
+
+    private final AddressSpaceView addressSpaceView;
+
+    @Getter
+    private final UUID streamId;
+
+    /**
+     * Last Address consumed/read by this stream
+     */
+    private volatile long lastAddressRead;
```

Comment:
let's replace to AtomicLong?

Reply:
This variable is used to coordinate between two threads, a thread that writes to it and another thread that reads from it. If I only need read-after write semantics and no CAS-related operations, why is it better to use an AtomicLong here?

Reply:
AtomicLong has same functionality, if you go inside that class you'll see there is a volatile variable inside the class.
Having AtomicLong instead of volatile indicates that the variable is volatile in compile time, and reduces the risk of using that variable incorrectly in the future. Next time another developer couldn't notice that the variable is volatile and would use it incorrectly, thinking that it is a regular variable, which is not possible in the case of atomic variables

Reply:
Since these volatiles are private and can only be changed within the class, I think that's not a strong argument to use a AtomicLong. Anyways, I changed it.